src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -579,7 +588,37 @@ bool JavaThread::is_interrupted(bool clear_interrupted) {
     java_lang_Thread::set_interrupted(threadObj(), false);
     WINDOWS_ONLY(osthread()->set_interrupted(false);)
   }
+  return interrupted;
+}
 
+// Checks and clears the interrupt status for platform or virtual thread.
+// Used by the JVMTI RawMonitorWait only.
+bool JavaThread::is_interrupted() {
```

Comment:
Reading code that uses JavaThread::is_interrupted() won't be clear that it resets the interrupt status. Can this be something that get_and_clear_interrupted or something that makes it clearer?

Reply:
Maybe `is_interrupted_with_reset`?  (though that would seem like an alias for `is_interrupted(true)`.

Reply:
Thanks, Alan and David. Will consider your suggestions.

Reply:
Renamed now.